src/unicorn/smt_solver.rs

```diff
@@ -489,7 +506,9 @@ pub mod z3solver_impl {
                     BV::new_const(self.context, name.to_owned(), width).into()
                 }
                 Node::Next { .. } => panic!("should be unreachable"),
-                Node::Bad { .. } => panic!("should be unreachable"),
+                Node::Bad { cond, .. } => {
```

Comment:
It would be better if we would directly referece the condition instead of referencing the `Bad` node in the OR'ed graph. That way `Bad` conceptually remains as not producing any output and the graph that smt_solver.rs sees is still purely combinatorial. We can clean this up later, and just add a TODO here in this regard.

Reply:
added!